cluster-autoscaler/integration/framework.go

```diff
@@ -245,38 +249,29 @@ func (driver *Driver) scaleWorkload(workloadName string, replicas int32) error {
 	return nil
 }
 
-func (driver *Driver) getFirstNode() (*v1.Node, error) {
+func (driver *Driver) getFirstAndLatestNode() (*v1.Node, *v1.Node, error) {
 	nodeList, err := driver.targetCluster.Clientset.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
 	if err != nil {
-		return &v1.Node{}, err
+		return nil, nil, err
 	}
-	sort.Slice(nodeList.Items, func(i, j int) bool {
-		return nodeList.Items[i].ObjectMeta.CreationTimestamp.Before(&nodeList.Items[j].ObjectMeta.CreationTimestamp)
-	})
-	firstNode := nodeList.Items[0]
-	return &firstNode, nil
-}
-
-func (driver *Driver) getLatestNode() (*v1.Node, error) {
-	nodeList, err := driver.targetCluster.Clientset.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
-	if err != nil {
-		return &v1.Node{}, err
-	}
-	sort.Slice(nodeList.Items, func(i, j int) bool {
-		return nodeList.Items[i].ObjectMeta.CreationTimestamp.Before(&nodeList.Items[j].ObjectMeta.CreationTimestamp)
-	})
 	if len(nodeList.Items) < 1 {
 		err = fmt.Errorf("no nodes found")
 	}
 	if err != nil {
-		return &v1.Node{}, err
+		return nil, nil, err
 	}
+	//sorting in ascending order of creation timeStamp
+	sort.Slice(nodeList.Items, func(i, j int) bool {
+		return nodeList.Items[i].ObjectMeta.CreationTimestamp.Before(&nodeList.Items[j].ObjectMeta.CreationTimestamp)
+	})
+	firstNode := nodeList.Items[0]
 	latestNode := nodeList.Items[len(nodeList.Items)-1]
-	return &latestNode, nil
+	return &firstNode, &latestNode, nil
 }
 
 func (driver *Driver) addAnnotationToNode(node *v1.Node) error {
-	(*node).ObjectMeta.Annotations[IGNORE_SCALEDOWN_ANNOTATION] = "true"
+	//test if this working else put back the * before node
```

Comment:
you can remove the comment if you have worked on it

Reply:
Ah! Yes, sure.

Reply:
done